pkg/storage/stores/shipper/compactor/deletion/delete_request.go

```diff
@@ -1,24 +1,31 @@
 package deletion
 
 import (
+	"github.com/grafana/loki/pkg/storage/stores/shipper/compactor/retention"
 	"github.com/prometheus/common/model"
 	"github.com/prometheus/prometheus/model/labels"
-	"github.com/prometheus/prometheus/promql/parser"
-
-	"github.com/grafana/loki/pkg/storage/stores/shipper/compactor/retention"
 )
 
-// DeleteRequest holds all the details about a delete request.
 type DeleteRequest struct {
-	RequestID string              `json:"request_id"`
-	StartTime model.Time          `json:"start_time"`
-	EndTime   model.Time          `json:"end_time"`
-	Selectors []string            `json:"selectors"`
-	Status    DeleteRequestStatus `json:"status"`
-	CreatedAt model.Time          `json:"created_at"`
+	RequestID    string              `json:"request_id"`
+	StartTime    model.Time          `json:"start_time"`
+	EndTime      model.Time          `json:"end_time"`
+	LogQLRequest string              `json:"logql_requests"`
```

Comment:
Maybe this could be a better name (or just `LogQL`)

```suggestion
	Query string              `json:"query"`
```

Reply:
Done manually, as it required a lot of renames